accounts-cluster-bench/src/main.rs

```diff
@@ -0,0 +1,549 @@
+use clap::{crate_description, crate_name, value_t, value_t_or_exit, App, Arg};
+use log::*;
+use rand::{thread_rng, Rng};
+use rayon::prelude::*;
+use solana_client::rpc_client::RpcClient;
+use solana_core::gossip_service::discover;
+use solana_faucet::faucet::{request_airdrop_transaction, FAUCET_PORT};
+use solana_measure::measure::Measure;
+use solana_sdk::rpc_port::DEFAULT_RPC_PORT;
+use solana_sdk::signature::{read_keypair_file, Keypair, Signature, Signer};
+use solana_sdk::timing::timestamp;
+use solana_sdk::{message::Message, transaction::Transaction};
+use solana_sdk::{system_instruction, system_program};
+use std::net::SocketAddr;
+use std::process::exit;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::{Arc, RwLock};
+use std::thread::{sleep, Builder, JoinHandle};
+use std::time::Duration;
+use std::time::Instant;
+
+pub fn airdrop_lamports(
+    client: &RpcClient,
+    faucet_addr: &SocketAddr,
+    id: &Keypair,
+    desired_balance: u64,
+) -> bool {
+    let starting_balance = client.get_balance(&id.pubkey()).unwrap_or(0);
+    info!("starting balance {}", starting_balance);
+
+    if starting_balance < desired_balance {
+        let airdrop_amount = desired_balance - starting_balance;
+        info!(
+            "Airdropping {:?} lamports from {} for {}",
+            airdrop_amount,
+            faucet_addr,
+            id.pubkey(),
+        );
+
+        let (blockhash, _fee_calculator) = client.get_recent_blockhash().unwrap();
+        match request_airdrop_transaction(&faucet_addr, &id.pubkey(), airdrop_amount, blockhash) {
+            Ok(transaction) => {
+                let mut tries = 0;
+                loop {
+                    tries += 1;
+                    let signature = client.send_transaction(&transaction).unwrap();
```

Comment:
In practice, I'm seeing intermittent panics on this line on `Transaction simulation failed: The bank has seen this signature before`. Presumably the transaction sent in the previous loop iteration is actually landing due to send-tx-service retries.

Reply:
I'm going to try polling for SingleGossip commitment, which seems sufficient and less likely to hit the 15s timeout:
```
let result = client.poll_for_signature_with_commitment(&signature, CommitmentConfig::single_gossip());
```